### PR TITLE
Added documentation and testing for object-registry.js

### PR DESCRIPTION
**Pull request by thefirebanks**
## Description
- Added unit testing for object-registry.js as per #143 
- Added documentation for object-registry.js as per #126 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/object-registry.js [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+ * @property {Object.<string, Protocol>} gc
+ */
 export default class ObjectRegistry {
   constructor() {
     this.objects = {};
     this.gc = {};
   }
 
   set(id, obj, gc = false) {
+    // We want to get rid of the current id and object in the objects map before replacing them
```
Comment (nahuakang) [inline src/object-registry.js:17]: Personally I'd start with an imperative sentence for one-line comments, like "Remove the current id and object..."

File: src/object-registry.js [hunk 2]
```diff
@@ -37,6 +45,10 @@ export default class ObjectRegistry {
     this.gc = {};
   }
 
+  /**
+   * Load the objects from an existing ObjectRegistry and store them in this.objects
```
Comment (nahuakang) [inline src/object-registry.js:49]: For method- and class-level documentation, I'd personally use third-person (which has also been the style in our repo so far :smiley:).

**Review by cereallarceny — APPROVED** [review]

**Comment by cereallarceny** [comment]
Nice work @thefirebanks. There are more issues if you want them!

File: src/object-registry.js [hunk 3]
```diff
@@ -1,12 +1,20 @@
 import * as tf from '@tensorflow/tfjs-core';
 
+/**
+ * ObjectRegistry stores a map of {id -> object} and a map of {id -> gc},
+    where gc denotes if we want the object of a given id to be garbage collected
+ *
+ * @property {Object.<string, Plan>} objects
+ * @property {Object.<string, Protocol>} gc
```
Comment (nahuakang) [inline src/object-registry.js:8]: One thing I haven't understood is what does `gc` stand for?

**Review by nahuakang — COMMENTED** [review]
Hey Daniel, looking great! You're ahead of me on unit tests haha :muscle: Made 3 tiny comments on the documentation part.

**Pull request merged**